File: 2582-pass-the-pillow/2582-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
        int num = 1;
        boolean turn = false;
        for(int i = 0; i < time; i++){
            if(!turn)
                num++;
            else
                num--;

            if(num>n){
                num = n-1;
                turn = true;
            }
            else if(num < 1){
                num = 2;
                turn = false;
            }
        }

        return num;
    }
}

File: 2582-pass-the-pillow/2582-pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        forward = True
        person = 1
        sec = 0

        while sec < time:
            if forward:
                person+=1
            else:
                person-=1
            
            if person == n or person==1:
                forward = not forward
            
            sec+=1
        
        return person